src/Common.cpp

```diff
@@ -177,13 +177,27 @@ std::vector<std::string> NAS2D::split(std::string str, char delim /*= ','*/, boo
 std::pair<std::string, std::string> NAS2D::splitOnFirst(const std::string& str, char delim)
 {
 	const auto delim_loc = str.find_first_of(delim);
-	return std::make_pair(str.substr(0, delim_loc), str.substr(delim_loc + 1));
+	if (delim_loc == std::string::npos)
+	{
+		return std::make_pair(std::string{}, str);
```

Comment:
I feel like this may be backwards. If there is no delimiter, and so nothing to split on, I would expect the full string in the first return value, while the second value was empty.

Reply:
I intended it to be "wherever the rest of the string is". In this case "the rest of the string" is the whole string. Maybe a comment or documentation?

Reply:
```cpp
splitOnFirst(",abc", ',');  // { "", "abc" }
splitOnFirst("a,bc", ',');  // { "a", "bc" }
splitOnFirst("ab,c", ',');  // { "ab", "c" }
splitOnFirst("abc,", ',');  // { "abc", "" }
splitOnFirst("abc", ',');  // { "abc", "" }  <- Suggestion (rather than { "", "abc" })
```
If the first component is all the stuff before the delimiter, and there is no delimiter, I feel the first component should then be the whole string.